### ISSUE TITLE
role="separator" not allowed on button via aria-allowed-role

### ISSUE THREAD

**Issue by gerardkcohen** [dequelabs/axe-core#4142]
### Product

axe-core

### Product Version

4.7.2

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
In order to achieve a simple, focusable separator I should be able to add `role="separator"` and `aria-valuenow` to a button.

#### Actual
Adding `role="separator"` to a button in order to achieve a focusable separator fails axe-core tests for aria-allowed-role as a best practice.

> To solve this problem, you need to fix the following:
> ARIA role separator is not allowed for given element

[ARIA role should be appropriate for the element](https://dequeuniversity.com/rules/axe/4.7/aria-allowed-role)

#### How to Reproduce
`<button role="separator" aria-valuenow="50" aria-label="Handle">></button>`

Sample code: [Simple focusable separator](https://output.jsbin.com/ciyisasude)

#### Additional context
Is there anything that is preventing me from overwriting the button semantics with `role="separator"` in order gain native focusability and event handling?

**Comment by straker** [comment]
Thanks for the issue. [ARIA in HTML](https://www.w3.org/TR/html-aria/) is the spec we follow when determining what roles and attributes are allowed on elements. According to the spec, the `button` element is [not allowed to have the role="separator"](https://www.w3.org/TR/html-aria/#el-button)

**Comment by gerardkcohen** [comment]
@straker Thanks for the response. I reached out and looks like that will be updated soon. See [separator role on button #488](https://github.com/w3c/html-aria/issues/488).

**Comment by straker** [comment]
Awesome. When https://github.com/w3c/html-aria/pull/489 is merged we can go ahead and add `separator` to the allowed list for button.

**Comment by scottaohara** [comment]
@straker I will be merging the ARIA in HTML PR today
